quantum/split_common/split_util.c

```diff
@@ -101,35 +101,46 @@ void keyboard_slave_loop(void) {
    #endif
 
    while (1) {
-      matrix_slave_scan();
-      
-      // read backlight info
+    // Matrix Slave Scan
+    matrix_slave_scan();
+    
+    // Read Backlight Info
     #ifdef BACKLIGHT_ENABLE
-        if (BACKLIT_DIRTY) {
-            backlight_set(i2c_slave_buffer[I2C_BACKLIT_START]);
-            BACKLIT_DIRTY = false;
-        }
+        #ifdef USE_I2C
+            if (BACKLIT_DIRTY) {
+                backlight_set(i2c_slave_buffer[I2C_BACKLIT_START]);
+                BACKLIT_DIRTY = false;
+            }
+        #else // USE_SERIAL
+            if (BACKLIT_DIRTY) {
+                backlight_set(serial_master_buffer[SERIAL_BACKLIT_START]);
```

Comment:
shouldn't `BACKLIT_DIRTY` be set to false after this, too?

Reply:
Yes, yes it should haha. This is why we have awesome reviewers like you ;)

Reply:
Also, wouldn't this be cleaner: 
```c
    #ifdef BACKLIGHT_ENABLE
            if (BACKLIT_DIRTY) {
              #ifdef USE_I2C
                backlight_set(i2c_slave_buffer[I2C_BACKLIT_START]);
              #else // USE_SERIAL
                backlight_set(serial_master_buffer[SERIAL_BACKLIT_START]);
              #endif
              BACKLIT_DIRTY = false;
            }
    #endif
```

I mean, either works, but...

Reply:
Psh "cleaner"

but yes you're correct, that's a better way to do it, will update haha

Reply:
Okay, "prettier" 😄